mars/web/tests/test_api.py

```diff
@@ -252,3 +253,12 @@ def testApi(self, mock_fetch_tensor, _):
             data_url = graph_url + '/data/' + c.key
             data = loads(requests.get(data_url).content)
             assert_array_equal(data, np.ones((100, 100)) * 100)
+
+            # test web session endpoint setter
+            self._web.stop()
+            self.start_web(self._scheduler_address)
```

Comment:
The same suggestion to use event, better to wrap the event and check logic to a class function to prevent from code copy.

Reply:
the same as the above comment.